src/main/java/chad/Ui.java

```diff
@@ -0,0 +1,138 @@
+package chad;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import chad.task.Deadline;
+import chad.task.Event;
+import chad.task.Task;
+
+/**
+ * Contains input methods for user to interact with Chadbot
+ */
+public class Ui {
+    /**
+     * Greets user when program runs
+     */
+    public static void greet() {
+        String startChat = "Hello! I'm Chadbot\nWhat can I do for you?\n";
+        startChat += "To see all available commands, type help";
+        Utility.printText(startChat);
+    }
+
+    /**
+     * Exit message when user close program
+     */
+    public static void closeChat() {
+        String exitChat = "Bye. Hope to see you again soon!";
+        Utility.printText(exitChat);
+        System.exit(0);
+    }
+
+    /**
+     * List all tasks in task list
+     * @param tasks arraylist of tasks
+     */
+    public static void listTask(ArrayList<Task> tasks) {
+        StringBuilder outputText = new StringBuilder();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            outputText.append(i + 1).append(".").append(task.toString()).append("\n");
+        }
+        Utility.printText(outputText.toString().trim());
+    }
+
+    /**
+     * Marks the task at specific index as done
+     * @param tasks arraylist of tasks
+     * @param taskID index of task
+     * @throws IOException throws error if file can't be open in chad.Storage
+     */
+    public static void markTask(ArrayList<Task> tasks, int taskID) throws IOException {
+        String outputText = "";
+        outputText += "Nice! I've marked this task as done:\n";
+        Task currentTask = tasks.get(taskID);
+        currentTask.markAsDone();
+        outputText += " " + currentTask;
+        Storage.toggleMarkTaskInFile(taskID);
+        Utility.printText(outputText);
+    }
+
+    /**
+     * Unmarks the task at specific index as done
+     * @param tasks arraylist of tasks
+     * @param taskID index of task
+     * @throws IOException throws error if file can't be open in chad.Storage
+     */
+    public static void unmarkTask(ArrayList<Task> tasks, int taskID) throws IOException {
+        String outputText = "";
+        outputText += "OK, I've marked this task as not done yet:\n";
+        Task currentTask = tasks.get(taskID);
+        currentTask.markAsUndone();
+        outputText += " " + currentTask;
+        Storage.toggleMarkTaskInFile(taskID);
+        Utility.printText(outputText);
+    }
+
+    public static void searchTaskByKeyword(ArrayList<Task> tasks, String userInput) {
+        String keyword = userInput.replaceFirst("find", "").trim();
+        String baseText = "Here are the matching tasks in your list:\n";
+        String output = baseText;
+        int count = 1;

```

Comment:
Perhaps you could consider changing the variable name to index instead of count, since the output would be printing the index of the task🤔